packages/lesswrong/components/ultraFeed/UltraFeedSettings.tsx

```diff
@@ -388,92 +557,91 @@ const UltraFeedSettings = ({
+    errors: {
+      lineClampNumberOfLines: zodErrors?.displaySettings?.lineClampNumberOfLines?._errors?.[0],
+      postTruncationBreakpoints: zodErrors?.displaySettings?.postTruncationBreakpoints,
+      commentTruncationBreakpoints: zodErrors?.displaySettings?.commentTruncationBreakpoints,
+    },
+    onLineClampChange: handleLineClampChange,
+    onBreakpointChange: handleBreakpointChange,
+  };
+  
+  const miscSettingsProps = {
+    formValues: {
+      incognitoMode: formValues.incognitoMode,
+      postTitlesAreModals: formValues.displaySetting.postTitlesAreModals,
+    },
+    onBooleanChange: handleBooleanChange,
 
-  const renderSimpleView = () => (
-    <>
-      <SourceWeightsSettings
-        weights={formValues.sourceWeights}
-        errors={sourceWeightErrors}
-        onChange={handleSourceWeightChange}
-      />
-      <TruncationGridSettings
-        levels={formValues}
-        onChange={handleTruncationLevelChange}
-        originalSettings={settings} 
-      />
-    </>
-  );
+  };
 
-  const renderAdvancedView = () => (
-    <>
-      <SourceWeightsSettings
-        weights={formValues.sourceWeights}
-        errors={sourceWeightErrors}
-        onChange={handleSourceWeightChange}
-      />
-      <MultipliersSettings
-        quickTakeBoost={{
-          value: formValues.quickTakeBoost,
-          error: getScalarError('quickTakeBoost'),
-          onChange: handleQuickTakeBoostChange,
-        }}
-        seenPenalty={{
-          value: formValues.ultraFeedSeenPenalty,
-          error: getScalarError('ultraFeedSeenPenalty'),
-          onChange: handleSeenPenaltyChange,
-        }}
-      />
-      <AdvancedTruncationSettings
-        onLineClampChange={handleLineClampChange}
-        onBreakpointChange={handleBreakpointChange}
-        values={{
-          lineClampNumberOfLines: formValues.lineClampNumberOfLines,
-          postBreakpoints: formValues.postBreakpoints,
-          commentBreakpoints: formValues.commentBreakpoints,
-        }}
-        errors={{
-          lineClampNumberOfLines: getScalarError('lineClampNumberOfLines'),
-          postBreakpoints: getArrayError('postTruncationBreakpoints'),
-          commentBreakpoints: getArrayError('commentTruncationBreakpoints'),
-        }}
-      />
-      <MiscSettings
-        formValues={formValues} 
-        onBooleanChange={handleBooleanChange}
-      />
-    </>
-  );
+  const exploreExploitBiasProps = {
+    currentLogImpactFactor: formValues.threadInterestModel.logImpactFactor,
+    onExploreBiasChange: handleExploreBiasChange,
+  };
 
   const hasAnyErrors = useMemo(() => {
      return zodErrors !== null;
   }, [zodErrors]);
 
+  const viewModeButton = (mode: 'simple' | 'advanced') => (
+    <div
+      onClick={() => setViewMode(mode)}
+      className={classNames(classes.viewModeButton, viewMode === mode ? classes.viewModeButtonActive : classes.viewModeButtonInactive)}
+    >
+      {mode.charAt(0).toUpperCase() + mode.slice(1)}
+    </div>
+  );
```

Comment:
I support factoring this out into a component but it shouldn't be defined inline